dagstore_test.go

```diff
@@ -322,19 +439,65 @@ func acquireShard(t *testing.T, dagst *DAGStore, k shard.Key, n int) {
 			if res.Error != nil {
 				return res.Error
 			}
-			defer res.Accessor.Close()
 
 			bs, err := res.Accessor.Blockstore()
 			if err != nil {
 				return err
 			}
 
-			_, err = bs.Get(rootCID)
-			return err
+			if _, err := bs.Get(rootCID); err != nil {
+				return err
+			}
+
+			mu.Lock()
+			accessors = append(accessors, res.Accessor)
+			mu.Unlock()
+			return nil
 		})
 	}
 
 	require.NoError(t, grp.Wait())
+	info := flushAndGetShardState(t, dagst, k)
+	require.Equal(t, ShardStateAvailable, info.ShardState)
+	require.NoError(t, info.Error)
+	// refs should be equal to number of acquirers since we've not closed any acquirer/released any shard.
+	require.EqualValues(t, n, info.refs)
+	// ensure transient file path exists.
+	abs := filepath.Join(dagst.config.TransientsDir, k.String())
+	_, err := os.Stat(abs)
+	require.NoError(t, err)
+
+	// close all the accessors
+	grp2, _ := errgroup.WithContext(context.Background())
+	for i := 0; i < n; i++ {
+		ac := accessors[i]
+		grp2.Go(func() error {
+			mu.RLock()
+			defer mu.RUnlock()
```

Comment:
Not needed at this point, as nobody is mutating the slice any longer.

Reply:
Therefore the whole thing can be simplified to:

```go
grp2.Go(ac.Close)
```

Reply:
@raulk We're accessing a loop variable in a go-routine here.